src/sql/magic_cmd.py

```diff
@@ -65,8 +66,95 @@ def execute(self, line="", cell="", local_ns=None):
 
             args = parser.parse_args(others)
             return inspect.get_columns(name=args.table, schema=args.schema)
+        elif cmd_name == "test":
+            parser = CmdParser()
+
+            parser.add_argument(
+                "-t", "--table", type=str, help="Table name", required=True
+            )
+            parser.add_argument(
+                "-c", "--column", type=str, help="Column name", required=False
+            )
+            parser.add_argument(
+                "-g",
+                "--greater",
+                type=str,
+                help="Greater than a certain number.",
+                required=False,
+            )
+            parser.add_argument(
+                "-goe",
+                "--greater-or-equal",
+                type=str,
+                help="Greater or equal than a certain number.",
+                required=False,
+            )
+            parser.add_argument(
+                "-l",
+                "--less-than",
+                type=str,
+                help="Less than a certain number.",
+                required=False,
+            )
+            parser.add_argument(
+                "-loe",
+                "--less-than-or-equal",
+                type=str,
+                help="Less than or equal to a certain number.",
+                required=False,
+            )
+            parser.add_argument(
+                "-nn",
+                "--no-nulls",
+                help="Returns rows in specified column that are not null.",
+                action="store_true",
+            )
+
+            args = parser.parse_args(others)
+            if args.greater and args.greater_or_equal:
+                return ValueError(
+                    "You cannot use both greater and greater "
+                    "than or equal to arguments at the same time."
+                )
+            elif args.less_than and args.less_than_or_equal:
+                return ValueError(
+                    "You cannot use both less and less than "
+                    "or equal to arguments at the same time."
+                )
+
+            query = construct_string_query(args)
+            conn = sql.connection.Connection.current.session
+            res = conn.execute(query).fetchall()
+            if args.no_nulls and len(res) > 0:
+                raise ValueError(
+                    "Specified column {} has null values present.".format(args.column)
+                )
+            return res
```

Comment:
they test functions should True when the data meets the conditions (i.e., not nulls), or raise an error when they not, instead of returning the results.

I think to make this more readable, the error displayed should contain more information. for example, it we run a "no NULLs" test, but there are nulls, we throw an exception and the error message should contain some of the records that contain NULLs so the user can investigate further

Reply:
to make the error messages more elaborate here's the approach i've theorized:

1. Seperate out the test into the list of arguments, for example strictly-greater-than, no-nulls etc.
2. Run the SQL query on each argument individually and store results (example column where it returns false, whether it's true or false)
3. Return either true if everything is true, or error message with examples of each argument that return false and an example.

Reply:
Let me know if this works.

Reply:
yeah, this makes sense.

if I understand correctly, you will convert the arguments that the user passes into a few queries, run them separately and then pick up whatever rows do not fullfill the conditions and show them?

sounds good